src/Perlang.Parser/PerlangParser.cs

```diff
@@ -564,7 +564,18 @@ private Expr UnaryPrefix()
             {
                 Token @operator = Previous();
                 Expr right = UnaryPrefix();
-                return new Expr.UnaryPrefix(@operator, right);
+
+                // We detect MINUS + NUMBER here and convert it to a single Expr.Literal() (with constant value) instead
+                // of retaining it as a unary prefix expression. See #302 for some more details about why this was
+                // changed.
+                if (@operator.Type == MINUS && right is Expr.Literal rightLiteral)
+                {
+                    return new Expr.Literal(NumberParser.MakeNegative(rightLiteral.Value!));
+                }
+                else
+                {
+                    return new Expr.UnaryPrefix(@operator, right);
+                }
```

Comment:
:point_up: This is an important aspect of this PR, where the logic in `NumberParser.MakeNegative` is part of solving #302 in a good way. (ensuring numbers like `-2147483648` are `System.Int32` and not `System.Int64`)